src/test/java/UseCases/UseCaseTest.java

```diff
@@ -20,12 +21,23 @@
     IngredientImpl ingredient1;
     IngredientImpl ingredient2;
 
+    RecipeImpl recipe1;
+    RecipeImpl recipe2;
+
+    TagStorageImpl tagStorage;
+
     @BeforeEach
     public void setup() {
```

Comment:
You can tell there was crunch time before phase 1. Why is there just one mega test class for our use cases? Each use case should have its own! As it's always been and always will be.